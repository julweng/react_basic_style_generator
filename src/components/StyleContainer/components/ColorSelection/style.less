@import "~styles/common.less";

.color__selection__container {
  display: flex;
  flex-direction: column;
  position: fixed;
  font-size: 1.25rem;
  font-family: @defaultFont;
  background-color: @backgroundColor;
  border-radius: @borderRadius;
  color: @textColor;
  width: 30%;
  border: none;
  z-index: 5;

  &.left {
    top: 28%;
    left: 9%;
  }

  &.right {
    top: 28%;
    right: 9%;
  }

  .color__selection {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    text-align: left;

    .background__selection > div:first-child,
    .text__selection > div:first-child {
      margin: 1rem;
    }

    .chrome-picker {
      width: 130% !important;
      box-shadow: rgba(73, 72, 72, 0.3) 0px 0px 1px,  rgba(73, 72, 72, 0.3) 0px 2px 4px;
      border-radius: @borderRadius;
    }
  }

  .submit__color__btn {
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin: 2rem;

    button:first-child {
      .selection-btn-mixin(@greyBtnColor, @textColor, @darkerWhite);
    }

    button:last-child {
      .selection-btn-mixin(@greenBtnColor, @backgroundColor, @darkerGreen);
    }
  }
}

@media screen and (max-width: 768px) {
  .color__selection__container {
    width: 40%;

    &.left {
      top: 26%;
      left: 5%;
    }
  
    &.right {
      top: 26%;
      right: 5%;
    }

    .color__selection {
      flex-direction: column;
      padding-left: 2rem;
      .chrome-picker {
        width: 120%;
      }
    }
    .submit__color__btn {
      flex-direction: column;
      button {
        width: 95% !important;
      }
    }
  }
}

@media screen and (max-width: 375px) {
  .color__selection__container {
    width: 85%;

    &.left {
      top: 21%;
      left: 7%;
    }
  
    &.right {
      top: 21%;
      right: 0;
      left: 7%;
    }

    .color__selection {
      flex-direction: column;
      padding-left: 2.5rem;
      .chrome-picker {
        width: 120%;
      }
    }
    .submit__color__btn {
      flex-direction: column;
      button {
        width: 95% !important;
      }
    }
  }
}
