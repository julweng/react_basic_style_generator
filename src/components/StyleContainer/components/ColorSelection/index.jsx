import React, { useState } from "react"
import { func, string, shape } from "prop-types"
import { CompactPicker } from "react-color"
import "./style.less"

const ColorSelection = ({ side, close, color, updateColor }) => {
  const [background, setBackground] = useState(color.backgroundColor)
  const [textColor, setTextColor] = useState(color.color)

  const handleBackgroundChange = color => setBackground(color)
  const handleTextColorChange = color => setTextColor(color)

  const handleSubmitColor = () => {
    updateColor(side, textColor.hex, background.hex)
    close()
  }

  return (
    <div className={`color__selection__container ${side}`}>
      <div className="color__selection">
        <div className="background__selection">
          <div>Background</div>
          <CompactPicker color={background} onChange={handleBackgroundChange} />
        </div>
        <div className="text__selection">
          <div>Text</div>
          <CompactPicker color={textColor} onChange={handleTextColorChange} />
        </div>
      </div>
      <div className="submit__color__btn">
        <button onClick={close}>Cancel</button>
        <button onClick={handleSubmitColor}>Submit</button>
      </div>
    </div>
  )
}

ColorSelection.propTypes = {
  close: func,
  updateColor: func,
  color: shape({
    color: string,
    backgroundColor: string
  }),
  side: string
}

export default ColorSelection
