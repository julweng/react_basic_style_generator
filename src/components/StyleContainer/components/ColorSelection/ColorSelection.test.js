import React from "react"
import { render, cleanup } from "@testing-library/react"
import "jest-dom/extend-expect"
import ColorSelection from "./index"

jest.mock("react-color", () => {
  // eslint-disable-next-line react/prop-types
  const CompactPicker = ({ color }) => (
    <div className="CompactPickerMock">{color}</div>
  )
  return { CompactPicker }
})

const createProps = ({
  side = "left",
  color = {
    color: "#000",
    backgroundColor: "#FFF"
  }
} = {}) => ({
  side,
  color,
  close: jest.fn(),
  updateColor: jest.fn()
})

describe("ColorSelection", () => {
  afterEach(cleanup)

  it("Should render without crashing", () => {
    const props = createProps()
    const { container } = render(<ColorSelection {...props} />)
    expect(container).toBeTruthy()
    expect(container).toMatchSnapshot()
  })

  it("Should render color picker labels", () => {
    const props = createProps()
    const { getByText } = render(<ColorSelection {...props} />)
    expect(getByText("Background")).toBeTruthy()
    expect(getByText("Text")).toBeTruthy()
  })

  it("Should render an instance of CompactPicker showing background color", () => {
    const props = createProps()
    const { container } = render(<ColorSelection {...props} />)
    const bgColorPicker = container.querySelectorAll(".CompactPickerMock")[0]
    expect(bgColorPicker).toHaveTextContent(props.color.backgroundColor)
  })

  it("Should render an instance of CompactPicker showing text color", () => {
    const props = createProps()
    const { container } = render(<ColorSelection {...props} />)
    const textColorPicker = container.querySelectorAll(".CompactPickerMock")[1]
    expect(textColorPicker).toHaveTextContent(props.color.color)
  })

  it("Should render a cancel button that closes the selection container on click", () => {
    const props = createProps()
    const { getByText } = render(<ColorSelection {...props} />)
    const cancelBtn = getByText("Cancel")
    cancelBtn.click()
    expect(props.close).toHaveBeenCalledTimes(1)
  })

  it("render a submit button that submits colro and closes the selection container on click", () => {
    const props = createProps()
    const { getByText } = render(<ColorSelection {...props} />)
    const submitBtn = getByText("Submit")
    submitBtn.click()
    expect(props.close).toHaveBeenCalledTimes(1)
    expect(props.updateColor).toHaveBeenCalledTimes(1)
  })
})
