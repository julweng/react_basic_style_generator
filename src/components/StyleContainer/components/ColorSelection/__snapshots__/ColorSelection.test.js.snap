// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelection Should render without crashing 1`] = `
<div>
  <div
    class="color__selection__container left"
  >
    <div
      class="color__selection"
    >
      <div
        class="background__selection"
      >
        <div>
          Background
        </div>
        <div
          class="CompactPickerMock"
        >
          #FFF
        </div>
      </div>
      <div
        class="text__selection"
      >
        <div>
          Text
        </div>
        <div
          class="CompactPickerMock"
        >
          #000
        </div>
      </div>
    </div>
    <div
      class="submit__color__btn"
    >
      <button>
        Cancel
      </button>
      <button>
        Submit
      </button>
    </div>
  </div>
</div>
`;
